runtime/jvmti/jvmtiHelpers.cpp

```diff
@@ -168,6 +168,13 @@ getVMThread(J9VMThread *currentThread, jthread thread, J9VMThread **vmThreadPtr,
 		if (NULL != carrierThread) {
 			targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, carrierThread);
 		}
+		if (-1 > J9OBJECT_I64_LOAD(currentThread, threadObject, vm->virtualThreadInspectorCountOffset)) {
+			/* If the virtual thread is suspended in transition, check if the mounting process is already completed. */
+			J9VMThread *carrierVMThread = (J9VMThread *)(J9OBJECT_U64_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset) & ~0x1);
+			if (NULL != carrierVMThread->currentContinuation) {
+				targetThread = carrierVMThread;
+			}
+		}
```

Comment:
This shouldn't be possible since an inTransition suspended thread should have the halt flag set.

Reply:
if the vthread is suspended after entering critical section, then it is possible that the carrierThread field on vthread have not been set, but the halt flag is set on the carrier as it is halted by the `internalSuspendState`. so the only access to carrier vm thread would be through the `internalSuspendState` field